### ISSUE TITLE
Per-pixel lighting for shaders

### ISSUE THREAD

**Issue by dylanede** [three-rs/three#233]
The Lambert and Phong shaders calculate lighting per-vertex. This results in poor lighting for meshes with low vertex density. There really is no need for this on modern hardware.

**Comment by kvark** [comment]
This is not true. Phong shaders are evaluating lighting on per-pixel basis: https://github.com/three-rs/three/blob/78e053448dfa213fcf432eaaf9b15d93259d10fb/data/shaders/phong_ps.glsl#L36

Is there a particular quality issue you are concerned about? Or just sharing the observation?
